sound/soc/sof/ipc4-topology.c

```diff
@@ -535,13 +535,19 @@ static void sof_ipc4_widget_free_comp_dai(struct snd_sof_widget *swidget)
 	case snd_soc_dapm_dai_in:
 	case snd_soc_dapm_dai_out:
 	{
+		struct sof_ipc4_available_audio_format *available_fmt;
 		struct snd_sof_dai *dai = swidget->private;
 		struct sof_ipc4_copier *ipc4_copier;
 
 		if (!dai)
 			return;
 
 		ipc4_copier = dai->private;
+		available_fmt = &ipc4_copier->available_fmt;
+
+		kfree(available_fmt->dma_buffer_size);
+		kfree(available_fmt->base_config);
```

Comment:
Dual question to my other comment: what does the notion of DMA buffer size mean for a DAI? I can see it for PCM devices with a ring buffer, I just don't see it for a DAI.

Reply:
> Dual question to my other comment: what does the notion of DMA buffer size mean for a DAI? I can see it for PCM devices with a ring buffer, I just don't see it for a DAI.

The dma_buffer_size is used for gateway configuration and it is "Preferred Gateway DMA buffer size (in bytes)"